backend/src/services/group_service.py

```diff
@@ -45,14 +45,22 @@ def fetch_scores_by_group(self, token):
             # Turn score from list of tuples into dict
             # e.g. [(50, 1), (75, 2), ..] =>
             # {1: 50, 2: 75, .. }
+            # And get amount of how many responses per id there is
             final_score = {}
-            for score, profile_id in scores:
+            response_ids = []
+            for score, profile_id, response_id in scores:
                 if profile_id not in final_score:
                     final_score[profile_id] = score
+                    if response_id not in response_ids:
+                        response_ids.append(response_id)
                 else:
                     final_score[profile_id] += score
+                    if response_id not in response_ids:
+                        response_ids.append(response_id)
```

Comment:
We can use [`set`](https://docs.python.org/3/library/stdtypes.html#set-types-set-frozenset) to simplify the unique response counting.

```suggestion
            # Use set to count unique response_ids
            response_ids = set()
            for score, profile_id, response_id in scores:
                response_ids.add(response_id)
                
                if profile_id not in final_score:
                    final_score[profile_id] = score
                else:
                    final_score[profile_id] += score
```